### PR TITLE
Update ruff to 0.7.0

### PR DESCRIPTION
**Pull request by cyyever**
### Description

Use latest ruff

### CODE REVIEW

**Comment by justinchuby** [comment]
Thanks, could you fix dco?

**Comment by cyyever** [comment]
@justinchuby Fixed

**Review by justinchuby — APPROVED** [review]

File: onnx/backend/test/runner/__init__.py [hunk 1]
```diff
@@ -232,21 +232,21 @@ def download_model(
         # On Windows, NamedTemporaryFile can not be opened for a
         # second time
         del model_dir
-        download_file = tempfile.NamedTemporaryFile(delete=False)
```
Comment (justinchuby) [inline onnx/backend/test/runner/__init__.py:235]: with delete being false, do we still need to use it as a context manager?

Comment (cyyever) [reply onnx/backend/test/runner/__init__.py:235]: To make ruff happy now

File: onnx/backend/test/runner/__init__.py [hunk 2]
```diff
@@ -232,21 +232,21 @@ def download_model(
...
-            )
-            urlretrieve(model_test.url, download_file.name)
-            print("Done")
-            onnx.utils._extract_model_safe(download_file.name, models_dir)
-        except Exception as e:
-            print(f"Failed to prepare data for model {model_test.model_name}: {e}")
-            raise
-        finally:
-            os.remove(download_file.name)
+        with tempfile.NamedTemporaryFile(delete=False) as download_file:
```
Comment (justinchuby) [inline onnx/backend/test/runner/__init__.py:235]: actually, do we need delete=False if there is `os.remove` in the final branch?

Comment (cyyever) [reply onnx/backend/test/runner/__init__.py]: Not need.

File: onnx/backend/test/runner/__init__.py [hunk 3]
```diff
@@ -232,21 +232,21 @@ def download_model(
...
-            )
-            urlretrieve(model_test.url, download_file.name)
-            print("Done")
-            onnx.utils._extract_model_safe(download_file.name, models_dir)
-        except Exception as e:
-            print(f"Failed to prepare data for model {model_test.model_name}: {e}")
-            raise
-        finally:
-            os.remove(download_file.name)
+        with tempfile.NamedTemporaryFile(
```
Comment (justinchuby) [inline onnx/backend/test/runner/__init__.py:235]: Looking at this again. I think a temp dir is probably more appropriate

File: onnx/backend/test/runner/__init__.py [hunk 4]
```diff
@@ -232,21 +232,21 @@ def download_model(
...
+            delete=True, delete_on_close=False
+        ) as download_file:
+            try:
+                download_file.close()
```
Comment (justinchuby) [inline onnx/backend/test/runner/__init__.py:239]: The file itself is never used. Creating a file in a tempdir seems to be more readable

File: onnx/backend/test/runner/__init__.py (hunk 3 above)
Comment (cyyever) [reply onnx/backend/test/runner/__init__.py]: Use it

File: onnx/backend/test/runner/__init__.py [hunk 5]
```diff
@@ -226,27 +226,21 @@ def assert_similar_outputs(
     def download_model(
         cls,
         model_test: TestCase,
-        model_dir: str,
```
Comment (justinchuby) [inline onnx/backend/test/runner/__init__.py:229]: Is this a public interface? Should we make kwarg only?

Comment (cyyever) [reply onnx/backend/test/runner/__init__.py:229]: No, I think it's a test class. It should be safe to break BC.

Comment (justinchuby) [reply onnx/backend/test/runner/__init__.py:229]: That would be ideal, but I’m not sure realistically what’s the state of usage. @xadupre does this look good to you?

**Review by justinchuby — APPROVED** [review]

**Pull request merged**